src/compiler/performance.ts

```diff
@@ -1,146 +1,174 @@
 /*@internal*/
 /** Performance measurements for the compiler. */
-namespace ts.performance {
-    let perfHooks: PerformanceHooks | undefined;
-    // when set, indicates the implementation of `Performance` to use for user timing.
-    // when unset, indicates user timing is unavailable or disabled.
-    let performanceImpl: Performance | undefined;
-
-    export interface Timer {
+namespace ts {

```

Comment:
Can you please summarize these changes?

Reply:
Basically before this change ts.performance.mark enable etc worked as namespace and recorded the timings. With tsc --build i needed to have two performance registries,.. one for project thats getting build (the per program one that we enable each time program starts building and disabled once thats done) and one for tsc --build related work measurement that spans through all projects building. Eg. calculating which project to invalidate etc or some calculations when writing output files etc.